core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -313,6 +313,7 @@ private object SpecialLengths {
   val PYTHON_EXCEPTION_THROWN = -2
   val TIMING_DATA = -3
   val END_OF_STREAM = -4
+  val NULL = -5
```

Comment:
Offline conversation: This change is related to the SPARK-1630 which was attempted to be solved in #1551 Since that PR was rejected by @mateiz its better that he takes a look and comment on the validity of this patch.